sdk/core/core-rest-pipeline/src/retryStrategies/throttlingRetryStrategy.ts

```diff
@@ -2,54 +2,77 @@
 // Licensed under the MIT license.
 
 import { PipelineResponse } from "..";
+import { parseHeaderValueAsNumber } from "../util/helpers";
 import { RetryStrategy } from "./retryStrategy";
 
 /**
- * Returns the number of milliseconds to wait based on a Retry-After header value.
- * Returns undefined if there is no valid value.
- * @param headerValue - An HTTP Retry-After header value.
+ * The headers that come back from Azure services representing
+ * the amount of time (minimum) to wait to retry (in milliseconds).
  */
-function parseRetryAfterHeader(headerValue: string): number | undefined {
+const RetryAfterMillisecondsHeaders: string[] = ["retry-after-ms", "x-ms-retry-after-ms"];
+/**
+ * The header that comes back from Azure services representing
+ * the amount of time (minimum) to wait to retry (in seconds or timestamp after which we can retry).
+ */
+const RetryAfterHeader = "Retry-After";
+
+/**
+ * A response is a throttling retry response if it has a throttling status code (429 or 503),
+ * as long as one of the [ "Retry-After" or "retry-after-ms" or "x-ms-retry-after-ms" ] headers has a valid value.
+ *
+ * Returns the `retryAfterInMs` value if the response is a throttling retry response.
+ * If not throttling retry response, returns `undefined`.
+ *
+ * @internal
+ */
+function getRetryAfterInMs(response?: PipelineResponse): number | undefined {
+  if (!(response && [429, 503].includes(response.status))) return undefined;
   try {
-    const retryAfterInSeconds = Number(headerValue);
-    if (!Number.isNaN(retryAfterInSeconds)) {
-      return retryAfterInSeconds * 1000;
-    } else {
-      // It might be formatted as a date instead of a number of seconds
+    let retryAfterValue = undefined;
```

Comment:
seems like we can move this declaration into the for loop since it's not used outside of it? Also allows it to be const.